terraform/modules/rds/main.tf

```
@@ -0,0 +1,56 @@
+# Non-compliant RDS configuration (missing encryption, backups, and monitoring)
```

The RDS instance is not compliant with SOC2 requirements as it lacks encryption, sufficient backup retention, and monitoring. Additionally, hardcoded credentials and public accessibility are security risks.

```suggestion
resource "aws_db_instance" "main" {
  identifier              = "${var.environment}-db"
  engine                 = "postgres"
  engine_version         = "13.7"
  instance_class         = var.instance_class
  allocated_storage      = 20

  name                   = var.db_name
  username               = var.db_username
  password               = var.db_password

  publicly_accessible    = false
  storage_encrypted      = true
  backup_retention_period = 30
  monitoring_interval    = 60
  performance_insights_enabled = true
  multi_az               = true

  vpc_security_group_ids = [aws_security_group.db.id]

  tags = {
    Name        = "${var.environment}-db"
    Environment = var.environment
  }
}
```